### PR TITLE
Detailpage

### PR DESCRIPTION
**Pull request by FabianAlleblas**
I added a detail page to my app, also a new input for additional informations.
The entity plant got a new property "info" to handle the new input data, nothing major in php, just doctrine doin' it's thing..

### CODE REVIEW

File: frontend/src/Components/Forms/AddPlantForm.js [hunk 1]
```diff
@@ -11,18 +11,19 @@ export default function AddPlantForm({ savePlantData }) {
 
   const history = useHistory()
   const initialPlant = {
-    plantname: '',
-    plantspecies: '',
+    name: '',
+    species: '',
+    info: '',
```
Comment (omorgenroth) [inline frontend/src/Components/Forms/AddPlantForm.js:16]: Do you really need that `initialplant` ?

File: frontend/src/Components/Header/Header.js [hunk 2]
```diff
@@ -1,37 +1,50 @@
 import PropTypes from 'prop-types'
 import styled from 'styled-components/macro'
+import { ReactComponent as ArrowIcon } from '../../assets/icons/arrowicon.svg'
 
-export default function Header({ className, children, secondaryStyle }) {
+export default function Header({
+  className,
+  children,
+  secondaryStyle,
+  hasArrow,
```
Comment (omorgenroth) [inline frontend/src/Components/Header/Header.js:9]: `hasArrow` is not very precise. What it acually means (especially outside of the component)would be something like "showBackButton" or "isReturnAllowed" etc.
Maybe consider renaming it, to make it more specific.

File: frontend/src/Components/Plantdetailcard/PlantDetailCard.js [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+export default function PlantDetailCard({ species, info, className }) {
+  PlantDetailCard.propTypes = {
+    species: PropTypes.string.isRequired,
+    info: PropTypes.string.isRequired,
+    className: PropTypes.string,
+  }
+  return (
+    <CardWrapper className={className}>
+      <ImageContainer src={monsteraImage}></ImageContainer>
```
Comment (ninapeters) [inline frontend/src/Components/Plantdetailcard/PlantDetailCard.js:13]: You don't embed anything in the `ImageContainer` so you don't need your the closing tag. Maybe something like `ImageFrame` would be more appropriate.

File: frontend/src/Components/Plantdetailcard/PlantDetailCard.js [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+`
+const TextContainer = styled.div`
+  text-align: center;
+`
+const PlantName = styled.h2`
+  color: var(--primary-plant-font-color);
+  font-size: 1.5rem;
+  font-weight: 700;
+`
+
```
Comment (ninapeters) [inline frontend/src/Components/Plantdetailcard/PlantDetailCard.js:44]: You can delete the blank line.

File: frontend/src/Components/Plantdetailcard/PlantDetailCard.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  background-size: cover, contain;
+  border-radius: 50%;
+  height: 200px;
+  width: 200px;
+`
+const TextContainer = styled.div`
+  text-align: center;
+`
+const PlantName = styled.h2`
+  color: var(--primary-plant-font-color);
```
Comment (ninapeters) [inline frontend/src/Components/Plantdetailcard/PlantDetailCard.js:40]: Do you need the `plant` in the name?

File: frontend/src/Pages/DetailPage/DetailPage.js [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+        {plant?.name}
+      </FixedHeader>
+      <CardContainer>
+        <PlantDetailCard species={plant?.species} info={plant?.info} />
+      </CardContainer>
+    </>
+  )
+
+  function useQuery() {
+    return new URLSearchParams(useLocation().search)
```
Comment (omorgenroth) [inline frontend/src/Pages/DetailPage/DetailPage.js:26]: nice :)

File: frontend/src/Pages/DetailPage/DetailPage.js [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+  const plant = plantList.find((plant) => plant.id === plantId)
+
+  return (
+    <>
+      <FixedHeader hasArrow onClick={openPlantList}>
+        {plant?.name}
+      </FixedHeader>
+      <CardContainer>
+        <PlantDetailCard species={plant?.species} info={plant?.info} />
```
Comment (ninapeters) [inline frontend/src/Pages/DetailPage/DetailPage.js:20]: 😎

**Review by ninapeters — COMMENTED** [review]
Looks very good. I left only a few comments.

**Comment by omorgenroth** [comment]
Great work!  I really like your structure. Made it relatively easy to understand the code.

**Review by omorgenroth — APPROVED** [review]
Great work!  I really like your structure. Made it relatively easy to understand the code.

File: frontend/src/Pages/Plantlistpage/PlantListPage.js [hunk 8]
```diff
@@ -12,15 +12,23 @@ export default function PlantListPage({ plantList }) {
       <FixedHeader>My Plants</FixedHeader>
       <ItemContainer>
         {plantList.map(({ id, plantname, plantspecies }) => (
-          <PlantCard key={id} name={plantname} species={plantspecies} />
+          <PlantCard
+            key={id}
+            name={plantname}
+            species={plantspecies}
+            onClick={() => openDetail({ id })}
```
Comment (marrep) [inline frontend/src/Pages/Plantlistpage/PlantListPage.js]: You could alternatively use the NavLink component of React instead of a onClick Listener

**Pull request merged**